packages/@aws-cdk/core/test/fs/fs-fingerprint.test.ts

```diff
@@ -179,6 +178,48 @@ describe('fs fingerprint', () => {
     });
   });
 
+  describe('fingerprint cache', () => {
+    const largeString = ' '.repeat(16 * 1024 * 1024);
+    const largefile = path.join(__dirname, 'inode-fp.1');
+
+    // Create a very large test file
+    beforeAll(() => {
+      const file = fs.openSync(largefile, 'w');
+      // 512MB file
+      for (let i = 0; i < 32; i++) {
+        fs.writeSync(file, largeString);
+      }
+      fs.closeSync(file);
+    });
+
+    afterAll(() => {
+      fs.unlinkSync(largefile);
+    });
+
+    test('caches fingerprint results', () => {
+      const openSyncSpy = jest.spyOn(fs, 'openSync');
+      const hash1 = FileSystem.fingerprint(largefile, {});
+      const hash2 = FileSystem.fingerprint(largefile, {});
+
+      expect(hash1).toEqual(hash2);
+      expect(openSyncSpy).toHaveBeenCalledTimes(1);
```

Comment:
You should restore the mock or clear all mocks in a `beforeEach()`. Because it is not cleared you end up with 3 calls in your `considers mtime` test. If this test is moved up in the file for some reason it will fail (test should not rely on testing order).

Also, is there a reason to create a large file for this test? It is the caching mechanism that is tested so it should work on files of any size?

```suggestion
      expect(openSyncSpy).toHaveBeenCalledTimes(1);
      
      openSyncSpy.mockRestore()
```

same for the second test

Reply:
Good point, we don't need to use a large file now that this is not sensitive to timing or filesize thresholds. I'll update that as well as adding a mock restore call.